app/lib/osm-p2p.js

```diff
@@ -64,9 +72,46 @@ function osmp2p() {
     return osm.query(q, opts, cb);
   }
 
+  function queryStream(q, opts) {
+    return osm.queryStream(q, opts);
+  }
+
+  function queryGeoJSONStream(q, opts) {
+    var osmStream = osm.queryStream(q, opts);
+    var geoJSONStream = getGeoJSON.obj(osm);
+    return pump(osmStream, geoJSONStream);
+  }
+
   function replicate(opts) {
     return osm.log.replicate(opts);
   }
+
+  function sync(transportStream, opts, callback) {
+    if (typeof opts === "function") {
+      callback = opts;
+      opts = null;
+    }
+
+    console.log("sync: start");
+    var osmStream = replicate(opts);
+
+    eos(osmStream, function(err) {
+      if (err) return console.log("osmStream error", err);
+      console.log("osm stream ended");
+      callback();
+    });
+    eos(transportStream, function(err) {
+      if (err) return console.log("transportStream error", err);
+      console.log("transport stream ended");
+      callback();
```

Comment:
That pattern is roughly what I had in an earlier version. However! `transportStream` wasn't ending and I wasn't sure why.